osu.Game/Beatmaps/BeatmapManager.cs

```diff
@@ -303,6 +307,11 @@ private List<BeatmapInfo> createBeatmapDifficulties(ArchiveReader reader)
                     // TODO: this should be done in a better place once we actually need to dynamically update it.
                     beatmap.BeatmapInfo.StarDifficulty = ruleset?.CreateInstance().CreateDifficultyCalculator(new DummyConversionBeatmap(beatmap)).Calculate().StarRating ?? 0;
 
+                    beatmap.BeatmapInfo.OnlineInfo = new BeatmapOnlineInfo
+                    {
+                        Length = beatmap.CalculateLength(),
```

Comment:
this shouldn't be in online info at this point.

Reply:
so if a property needs to be used locally, it should move out of `BeatmapOnlineInfo`, right?

Reply:
online info is specifically for content retrieved from an online source.